types/aws-lambda/trigger/cdk-custom-resource.d.ts

```diff
@@ -0,0 +1,65 @@
+import { CloudFormationCustomResourceEvent } from './cloudformation-custom-resource';
+import { Handler } from '../handler';
+
+// The CDK docs only specify 'important' properties, but in reality the incoming event
+// to the Lambda matches that of a traditional custom resource.
+// This includes the ResponseURL property which should not be used as the framework
+// itself will deal with delivering responses.
+export type CdkCustomResourceEvent = CloudFormationCustomResourceEvent & {
+    /**
+     * **This URL should not be used.** The CDK Provider Framework will call this URL
+     * automatically based on the response produced by the Lambda handler.
+     */
+    ResponseURL: string;
+};
```

Comment:
Nice trick to document usage! I might even suggest `Omit<CloudFormationCustomResourceEvent, "ResponseURL">`, but I prefer to be accurate to what actually exists... maybe someone can do something like debugging with it?

Reply:
Omit isn't a bad idea.
Currently this does seem to work okay under VS Code, but 'go to definition' correctly returns multiple definitions as a result:
![image](https://user-images.githubusercontent.com/7294642/139643615-d0dede4a-bb73-4c12-98be-815ada1ee328.png)

Reply:
Hmm, I tried Omit but it seemed to break the type union around `RequestType` and only allowing common properties. I'll leave this as-is.